File: daily_python/3838_Weighted_Word_Mapping.py
import os
import sys
from typing import List

class Solution:
    def mapWordWeights(self, words: List[str], weights: List[int]) -> str:
        return "".join([chr(ord('a') + 25 - sum(weights[ord(c) - ord('a')] for c in word) % 26) for word in words])
    
def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            words = list(lines[0].split('=')[1].strip()[2:-3].split('","'))
            weights = list(map(int, lines[1].split('=')[1].strip()[1:-2].split(',')))
            dataset.append((words, weights))
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for words, weights in dataset:
        results.append(solution.mapWordWeights(words, weights))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')

File: daily/2026/3838_Weighted_Word_Mapping.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    string mapWordWeights(vector<string>& words, vector<int>& weights) {
        string ans;
        for (string s : words){
            int acc = 0;
            for (char c : s){
                acc = (acc + weights[c - 'a']) % 26;
            }
            ans += ('a' + 25 - acc);
        }
        return ans;
    }
};

int main(){
    vector<string> words = {"abcd","def","xyz"};
    vector<int> weights = {5,3,12,14,1,2,3,2,10,6,6,9,7,8,7,10,8,9,6,9,9,8,3,7,7,2};
    // vector<string> words = {"a","b","c"};
    // vector<int> weights = {1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1};
    // vector<string> words = {"abcd"};
    // vector<int> weights = {7,5,3,4,3,5,4,9,4,2,2,7,10,2,5,10,6,1,2,2,4,1,3,4,4,5};
    Solution S;
    cout << S.mapWordWeights(words, weights) << endl;
    return 0;
}